src/shared/state/Applet/Applet.schema.ts

```diff
@@ -17,14 +17,24 @@ export type CreateAppletStateData = {
   key: keyof AppletSchema;
 };
 
+type ActivityFlowItem = {
+  activityId: string;
+  id: string;
+  order: number;
+  activityKey?: string;
```

Comment:
do we really need to use activityId and activityKey? maybe assign `activityKey` to activityId as a fallback as you did before in other places?